database/changesets/20210422_TerraDataRepoSourceDetailsTable.xml

```diff
@@ -0,0 +1,28 @@
+<?xml version="1.1" encoding="UTF-8" standalone="no"?>
+<databaseChangeLog
+        xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+        xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog-ext
+                        http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd
+                        http://www.liquibase.org/xml/ns/dbchangelog
+                        http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.8.xsd">
+    <changeSet author="chengche@broadinstitute.org"
+               dbms="postgresql"
+               id="20210422"
+               logicalFilePath="20210422_TerraDataRepoSourceDetailsTable.xml"
+               runInTransaction="false">
+        <comment>
+            the details table of 'Terra DataRepo' source
+        </comment>
+        <sql>
+            CREATE TYPE TerraDataRepoDetails AS (
+                id                          bigint,      -- unique to this table
+                snapshot_creation_job_id    text,        -- id of the TDR snapshot creation job
+                snapshot_id                 text,        -- id of the TDR snapshot
+                snapshot_reference_id       text,        -- id of the TDR snapshot reference
```

Comment:
This may be more accurately described as "id of the Terra Workspace snapshot reference": the snapshot lives in TDR and the snapshot reference lives in the workspace.